nipype/interfaces/dtitk/registration.py

```diff
@@ -1,394 +1,431 @@
-                >>> node.inputs.in_tensor = 'diffusion.nii'
-                >>> node.inputs.in_xfm = 'ants_Warp.nii.gz'
-                >>> node.run() # doctest: +SKIP
-                """
-
-    input_spec = diffeoSymTensor3DVolInputSpec
-    output_spec = diffeoSymTensor3DVolOutputSpec
-    _cmd = 'deformationSymTensor3DVolume'
-
-    def _list_outputs(self):
-        outputs = self.output_spec().get()
-        outputs['out_file'] = self.inputs.out_file
-        return outputs
-
 
 class affSymTensor3DVolInputSpec(CommandLineInputSpec):
-    in_tensor = traits.Str(desc='moving tensor', exists=True, mandatory=False,
-                           position=0, argstr="-in %s")
-    in_xfm = traits.Str(desc='transform to apply', exists=True,
-                        mandatory=False, position=1, argstr="-trans %s")
-    in_target = traits.Str(desc='', exists=True, mandatory=False, position=2,
-                           argstr="-target %s")
-    out_file = traits.Str(desc='', exists=True, mandatory=False, position=3,
-                          argstr="-out %s", name_source="in_tensor",
-                          name_template="%s_affxfmd.nii.gz")
+    in_file = File(desc='moving tensor volume', exists=True,
+                   argstr="-in %s", mandatory=True)
+    out_file = File(desc='output filename',
+                    argstr="-out %s", name_source="in_file",
+                    name_template="%s_affxfmd", keep_extension=True)
+    transform = File(exists=True, argstr="-trans %s",
+                     xor=['target', 'translation', 'euler', 'deformation'],
+                     desc='transform to apply: specify an input transformation\
+                      file; parameters input will be ignored',)
+    interpolation = traits.Enum('LEI', 'EI', usedefault=True,
+                                argstr="-interp %s",
+                                desc='Log Euclidean/Euclidean Interpolation')
+    reorient = traits.Enum('PPD', 'NO', 'FS', argstr='-reorient %s',
+                           usedefault=True, desc='Reorientation strategy: \
+                           preservation of principal direction, no \
+                           reorientation, or finite strain')
+    target = File(exists=True, argstr="-target %s", xor=['transform'],
+                  desc='output volume specification read from the target volume\
+                   if specified')
+    translation = traits.Tuple((traits.Float(), traits.Float(),
+                               traits.Float()),
+                               desc='translation (x,y,z) in mm',
+                               argstr='-translation %g %g %g',
+                               xor=['transform'])
+    euler = traits.Tuple((traits.Float(), traits.Float(), traits.Float()),
+                         desc='(theta, phi, psi) in degrees',
+                         xor=['transform'], argstr='-euler %g %g %g')
+    deformation = traits.Tuple(traits.Tuple((traits.Float(),) * 6, ...),
```

Comment:
Oh, sorry. I meant:

```Python
deformation = traits.Tuple((traits.Float(),) * 6, 
                           desc='(xx,yy,zz,xy,yz,xz)', xor=['transform'],
                           argstr='-deformation %g %g %g %g %g %g')
```

The `...` was just to avoid copying and pasting all the other stuff. :-)